subsys/bluetooth/mesh/settings.c

```diff
@@ -2474,6 +2355,48 @@ int bt_mesh_model_data_store(struct bt_mesh_model *mod, bool vnd,
 	return err;
 }
 
+void bt_mesh_store_rpl(void *entry)
+{
+	struct bt_mesh_rpl *rpl = (struct bt_mesh_rpl *)entry;
+	struct rpl_val val;
+	char path[18];
+	int err;
+
+	BT_DBG("src 0x%04x seq 0x%06x old_iv %u", rpl->src, rpl->seq,
+	       rpl->old_iv);
+
+	val.seq = rpl->seq;
+	val.old_iv = rpl->old_iv;
+
+	snprintk(path, sizeof(path), "bt/mesh/RPL/%x", rpl->src);
+
+	err = settings_save_one(path, &val, sizeof(val));
+	if (err) {
+		BT_ERR("Failed to store RPL %s value", log_strdup(path));
+	} else {
+		BT_DBG("Stored RPL %s value", log_strdup(path));
+	}
+}
```

Comment:
This change removes the staggered storing of replay entries, which may reduce the flash lifetime significantly. I think it would be better to not change the settings module for this patch, and expose the `rpl_tabel` as an extern in the rpl header, until we get around to fixing #19850. After this, we can retain the old functionality without exposing the internal data structure at all.